src/ray/object_manager/object_manager.cc

```diff
@@ -352,6 +355,10 @@ void ObjectManager::Push(const ObjectID &object_id, const ClientID &client_id) {
         for (uint64_t chunk_index = 0; chunk_index < num_chunks; ++chunk_index) {
           send_service_.post([this, client_id, object_id, data_size, metadata_size,
                               chunk_index, info]() {
+            // NOTE: When this callback executes, it's possible that the object
+            // will have already been evicted. It's also possible that the
+            // object could be in the process of being transferred to this
+            // object manager from another object manager.
```

Comment:
I see. Great observation. How are we handling this scenario? Is the check in `plasma/client.cc` no longer fatal?